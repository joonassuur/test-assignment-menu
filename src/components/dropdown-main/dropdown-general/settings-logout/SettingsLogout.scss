.settings-logout-container {
  border-bottom: 1px solid #dcdce0;
  .item {
    display: flex;
    align-items: center;
    .material-icons-outlined {
      margin-right: 5px;
    }
    &.exit {
      color: #D33649;
    }
  }
}
