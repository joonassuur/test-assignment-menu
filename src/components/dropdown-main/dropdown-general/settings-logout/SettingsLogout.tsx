import React from "react";
import { useDispatch } from "react-redux";
import { setDropdown } from "../../../../redux/Index";
import "./SettingsLogout.scss";

function SettingsLogout() {
  const dispatch = useDispatch();

  const closeDD = () => {
    dispatch(setDropdown(false));
  };
  return (
    <div className="settings-logout-container">
      <div className="item" onClick={closeDD}>
        <span className="material-icons-outlined">settings</span>
        <span>Settings</span>
      </div>
      <div className="item exit" onClick={closeDD}>
        <span className="material-icons-outlined">exit_to_app</span>
        <span>Log out</span>
      </div>
    </div>
  );
}

export default SettingsLogout;
