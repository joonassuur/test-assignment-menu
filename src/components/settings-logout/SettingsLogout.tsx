import React from "react";
import "./SettingsLogout.scss";

function SettingsLogout() {
  return (
    <div className="settings-logout-container">
      <div className="item">Settings</div>
      <div className="item">Log out</div>
    </div>
  );
}

export default SettingsLogout;
