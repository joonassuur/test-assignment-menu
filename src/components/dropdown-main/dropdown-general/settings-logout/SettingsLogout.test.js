import React from "react";
import TestRenderer from "react-test-renderer";

import SettingsLogout from "./SettingsLogout";

const mockDispatch = jest.fn();

jest.mock("react-redux", () => ({
  useSelector: jest.fn((fn) => fn()),
  useDispatch: () => mockDispatch,
}));

jest.mock("../../../../redux/Index");

describe("SettingsLogout component", () => {
  test("Matches the snapshot", () => {
    const settingsLogout = TestRenderer.create(<SettingsLogout />);
    expect(settingsLogout.toJSON()).toMatchSnapshot();
  });
});
