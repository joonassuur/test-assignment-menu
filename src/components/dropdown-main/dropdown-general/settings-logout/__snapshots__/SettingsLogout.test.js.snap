// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsLogout component Matches the snapshot 1`] = `
<div
  className="settings-logout-container"
>
  <div
    className="item"
    onClick={[Function]}
  >
    <span
      className="material-icons-outlined"
    >
      settings
    </span>
    <span>
      Settings
    </span>
  </div>
  <div
    className="item exit"
    onClick={[Function]}
  >
    <span
      className="material-icons-outlined"
    >
      exit_to_app
    </span>
    <span>
      Log out
    </span>
  </div>
</div>
`;
